# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/nsFrameLoader.cpp
# Commit: 6c5ad42069d0
# Full Hash: 6c5ad42069d09a8bc8d8306ab4b21c00230091ec
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2019-07-09 03:44:14
# Description:
#   Bug 1563542 - Correctly align usage of mIsDiscarded and mClosed for BrowsingContext, r=peterv
#   
#   In the bug which introduced mIsDiscarded, the code was changed to not set
#   mClosed during Detach, and only set mIsDiscarded. This was a mistake because a
#   bunch of places are only reading mClosed. Specifically when creating a
# ==============================================================================

diff -r db4c8cbf463e -r 6c5ad42069d0 dom/base/nsFrameLoader.cpp
--- a/dom/base/nsFrameLoader.cpp	Fri Jul 05 22:08:13 2019 +0000
+++ b/dom/base/nsFrameLoader.cpp	Mon Jul 08 17:27:27 2019 +0000
@@ -317,8 +317,8 @@
 
   RefPtr<BrowsingContext> parentContext = parentDocShell->GetBrowsingContext();
 
-  // Don't create a child docshell for a closed browsing context.
-  if (NS_WARN_IF(!parentContext) || parentContext->GetClosed()) {
+  // Don't create a child docshell for a discarded browsing context.
+  if (NS_WARN_IF(!parentContext) || parentContext->IsDiscarded()) {
     return nullptr;
   }
 